### PR TITLE
WIP: Migrate to Kotlin Gradle DSL

### PR DESCRIPTION
**Pull request by AOrobator**
The Kotlin Gradle DSL has groovy interoperability so it can be mixed and match with groovy. Therefore this PR can be merged as is, cuz the Android example is done, but ideally, the whole project should be converted.

https://docs.gradle.org/5.6.1/userguide/kotlin_dsl.html#sec:kotlin-dsl_plugin

@patjackson52

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -15,6 +15,11 @@ buildscript {
         classpath deps.plugins.android
     }
 }
+
+plugins {
+    id("de.fayard.buildSrcVersions") version "0.4.2"
+}
+
 apply from: "$rootDir/gradle/dependencies.gradle"
```
Comment (AOrobator) [inline build.gradle]: Eventually this line should be deleted

File: buildSrc/src/main/kotlin/Libs.kt [hunk 2]
```diff
@@ -0,0 +1,173 @@
...
+object Libs {
+  /**
+   * http://developer.android.com/tools/extras/support-library.html
+   */
+  const val appcompat: String = "androidx.appcompat:appcompat:" + Versions.appcompat
+
+  /**
+   * https://developer.android.com/testing
+   */
+  const val espresso_core: String = "androidx.test.espresso:espresso-core:" + Versions.espresso_core
```
Comment (patjackson52) [inline buildSrc/src/main/kotlin/Libs.kt]: I'm new to Kotlin Gradle DSL.  is it convention to use snakecase?  or just whatever you'd like?

**Comment by patjackson52** [comment]
lgtm, just the one comment.  I'd like the tests to pass.  Currently the tests only run on branches with names 'feature/*' or 'bugfix/*'

File: buildSrc/src/main/kotlin/Libs.kt (hunk 2 above)
Comment (AOrobator) [reply buildSrc/src/main/kotlin/Libs.kt]: Not really sure what the convention is for Kotlin Gradle DSL, but I can't imagine it should stray too far from Kotlin conventions. In either case, this is auto-generated code that shouldn't be edited. All hand written code should strictly follow Kotlin coding conventions.

**Comment by AOrobator** [comment]
Are the test runs documented in the readme? What happens when somebody forks this repo and makes a PR? Do the tests run then?

**Comment by AOrobator** [comment]
It might be easier/simpler to have tests run on all branches. I created https://github.com/reduxkotlin/redux-kotlin/pull/17 with the required branch names and am closing this one out.

**Pull request closed without merging**